### PR TITLE
drivers: wifi: Provide option to disable beamforming

### PR DESCRIPTION
**Pull request by sachinthegreen**
Provide a Kconfig option and driver API to disable beamforming.

### CODE REVIEW

**Review by VivekUppunda — APPROVED** [review]

**Review by udaynordic — APPROVED** [review]

File: drivers/wifi/nrf700x/Kconfig [hunk 1]
```diff
@@ -351,3 +351,7 @@ endif # NETWORKING
 config WIFI_MGMT_SCAN_SSID_FILT_MAX
 	default 2
 endif
+
+config NRF_WIFI_BEAMFORMING
+	bool "Enable Wi-Fi beamforming"
+	default y
```
Comment (ioannisg) [inline drivers/wifi/nrf700x/Kconfig:357]: Would you need some quick tradeoff explanation here?

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/radio_test/fmac_api.h [hunk 2]
```diff
@@ -99,8 +99,8 @@ enum nrf_wifi_status nrf_wifi_fmac_radio_test_prog_rx(struct nrf_wifi_fmac_dev_c
  * @param lna_gain LNA gain value.
  * @param bb_gain Baseband gain value.
  *
- * This function is used to send a command to:
- *	- The RPU firmware to start the RF test capture in radio test mode.
+ * This function is used to send a command to the RPU firmware to start the
```
Comment (krish2718) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/radio_test/fmac_api.h:102]: Please move unrelated changes to a separate commit.

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/radio_test/fmac_api.h [hunk 3]
```diff
@@ -245,11 +246,12 @@ void nrf_wifi_fmac_dev_rem_rt(struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx);
  */
 enum nrf_wifi_status nrf_wifi_fmac_dev_init_rt(struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx,
 #if defined(CONFIG_NRF_WIFI_LOW_POWER) || defined(__DOXYGEN__)
-			int sleep_type,
+					       int sleep_type,
```
Comment (krish2718) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/radio_test/fmac_api.h:249]: Please move unrelated changes to a separate commit.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c [hunk 4]
```diff
@@ -554,6 +555,7 @@ enum nrf_wifi_status nrf_wifi_fmac_dev_add_zep(struct nrf_wifi_drv_priv_zep *drv
 #endif /* CONFIG_NRF_WIFI_LOW_POWER */
 					NRF_WIFI_DEF_PHY_CALIB,
 					op_band,
+					CONFIG_NRF_WIFI_BEAMFORMING,
```
Comment (krish2718) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c:558]:
```suggestion
					IS_ENABLED(CONFIG_NRF_WIFI_BEAMFORMING),
```

**Review by krish2718 — CHANGES_REQUESTED** [review]
Also, please the reason for introducing this change in the commit log.

File: drivers/wifi/nrf700x/Kconfig (hunk 1 above)
Comment (sachinthegreen) [reply drivers/wifi/nrf700x/Kconfig:357]: Added.

**Comment by sachinthegreen** [comment]
> Also, please the reason for introducing this change in the commit log.

Updated commit log

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/radio_test/fmac_api.h (hunk 2 above)
Comment (sachinthegreen) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/radio_test/fmac_api.h]: Reverted this change.

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/radio_test/fmac_api.h (hunk 3 above)
Comment (sachinthegreen) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/radio_test/fmac_api.h:249]: This change is for adding a function parameter...the realignment is done along with that.

Comment (sachinthegreen) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/radio_test/fmac_api.h:249]: This change is for adding a new function parameter...the realignment is done along with the addition.
